class-activity-6.Rmd

```diff
@@ -0,0 +1,122 @@
+3) Calculates the mean of those clusters of values
+4) Selects the observation closest to the mean of the cluster
+5) Re-associates all observations closest to this observation
+6) Continues this process until the clusters are no longer changing
+
+Notice that in this case we have 5 variables and in class we only had 2. It is impossible to vizualise this process with 5 variables.
+
+Also, we need to choose the number of clusters we think are in the data. We will start with 4.
+
+```{r}
+
+fit_map <- kmeans(DF2, 2) 
+
+#We have created an object called "fit" that contains all the details of our clustering including which observations belong to each cluster.
+
+#We can access the list of clusters by typing "fit$cluster", the top row corresponds to the original order the rows were in. Notice we have deleted some rows.
+
+fit_map$cluster
+
+#We can also attach these clusters to te original dataframe by using the "data.frame" command to create a new data frame called K4.
+
+DF3 <- data.frame(DF2, fit_map$cluster)
+
+#Have a look at the K4 dataframe. Lets change the names of the variables to make it more convenient with the names() command.
+
+names(DF3) <- c("lon", "lat", "cluster") #c() stands for concatonate and it creates a vector of anything, in this case a vector of names.
+
+```
+
+# Visualize your clusters in ggplot
+```{r}
+#Create a scatterplot that plots location of each student and colors the points according to their cluster 
+
+g<-ggplot(data=DF3, aes(x=lon,y=lat, xlab = "longitude", ylab="latitude", colour=DF3$cluster)) + geom_point()
+g
+
+```
+
+# Can you group students according to their weekly activities?
+How much time did you spend this week in hours: - Coding (R, Java, SPSS, etc)
+How much time did you spend this week in hours: - Writing papers for class
+How much time did you spend this week in hours: - Reading/videos for class
+How much time did you spend this week in hours: - Watching videos for fun
+How much time did you spend this week in hours: - Cooking
+How much time did you spend this week in hours: - Using social media
+How much time did you spend this week in hours: - Social activities (movies, restaurants, etc)
+```{r}
+#Load data
+DF1 <- read.csv("HUDK405018-cluster.csv", header = TRUE, na.strings=c("","NA"), stringsAsFactors = FALSE)
+
+#Wrangle data
+DF1 <- select(DF1,Q2_1:Q2_7)
+DF2<-DF1[-c(1,2),]
+names(DF2) <- c("1", "2", "3","4","5","6", "7")
+
+# turn NAs into zeroes
+DF2[is.na(DF2)] <- 0
+DF3<-DF2
+#I know there's probably a faster way to do this but I got tired of searching for it
+DF3$'1' <-as.numeric(DF2$'1')
+DF3$'2' <-as.numeric(DF2$'2')
+DF3$'3'<-as.numeric(DF2$'3')
+DF3$'4'<-as.numeric(DF2$'4')
+DF3$'5'<-as.numeric(DF2$'5')
+DF3$'6'<-as.numeric(DF2$'6')
+DF3$'7'<-as.numeric(DF2$'7')
+
+#Rectifying the earlier stupid mistake to change names to numbers
+names(DF3)<-c("coding","writing papers","read/vid", "watch videos","cooking","social media", "going out")
+
```

Comment:
You probably don't need to as questions are all in the same unit. Scaling is when your variables are on different scales. K-means will be "sucked toward" the variable with the greater number of increments, essentially making all the other variables meaningless for the clusters.